src/setup-beam.js

```diff
@@ -560,38 +559,50 @@ function getRunnerOSVersion() {
   return containerFromEnvImageOS
 }
 
-async function get(url0, pageIdxs) {
-  async function getPage(pageIdx) {
-    const url = new URL(url0)
-    const headers = {}
-    const GithubToken = getInput('github-token', false)
-    if (GithubToken && url.host === 'api.github.com') {
-      headers.authorization = `Bearer ${GithubToken}`
-    }
+async function getUrlResponse(url, headers, attempt = 1) {
+  try {
+    const response = await fetch(url, { headers })
+    const contentType = response.headers.get('content-type') || ''
 
-    if (pageIdx !== null) {
-      url.searchParams.append('page', pageIdx)
+    if (!response.ok) {
+      throw new Error(`Got ${response.statusCode} from ${url}`)
     }
 
-    const httpClient = new http.HttpClient('setup-beam', [], {
-      allowRetries: true,
-      maxRetries: 3,
-    })
-    const response = await httpClient.get(url, headers)
-    if (response.statusCode >= 400 && response.statusCode <= 599) {
-      throw new Error(
-        `Got ${response.statusCode} from ${url}. Exiting with error`,
-      )
+    if (contentType.indexOf('application/json') !== -1) {
+      return response.json()
+    } else {
+      return response.text()
+    }
+  } catch (err) {
+    if (attempt <= MAX_HTTP_RETRIES) {
+      core.debug(`Error during fetch. Retrying in 1000ms: ${err}`)
+      await new Promise((resolve) => setTimeout(resolve, 1000))
```

Comment:
Extremely simple backoff mechanic. I'm not sure we need to get too overly complicated here.

Reply:
I'd do exponential backoff with the timeout. You already have `attempt` there so it's a question of using some simple math.

Reply:
Added 👍